buildkitd/oom-adjust.sh.template

```diff
@@ -0,0 +1,10 @@
+#! /bin/sh
+
+if [ "${OOM_SCORE_ADJ}" -eq "0" ]; then
+    exit 0
+fi
+
+for PID in $(pidof buildkit-runc)
+do
+    echo ${OOM_SCORE_ADJ} > /proc/$PID/oom_score_adj
+done
```

Comment:
Wondering if this script should be moved to the buildkit repo so that it reduces coupling between earthly and buildkit.

No strong opinion though, if that is not easy to do.

Reply:
It doesn't seem like it would reduce coupling; more like move it. Our entrypoint script, at that point, would be templating a file that doesn't exist in this repo; rather than buildkit calling a file that doesn't exist in its repo.

We could take some extra steps and do it "more right", where we have a script wrapper around buildkit that templates this script, and we invoke that wrapper instead of buildkit? But is that wrapping too much?

Reply:
Vlad, I think you are right; now that I think deeper. This script template should move over there; and I should include a `COPY` to bring it in in our buildkit Earthfile (to avoid having merges on the Dockerfile).

I'll also add a check for if the variable is set to this script, so regular builds with the hook will still work too.

Reply:
Another way to reduce coupling might be to make the hook configurable via an env var, and have that env var + this script (which would be referenced by that env var) be passed in our entrypoint.

Reply:
I could make the hooks configurable via buildkit config... is it worth carrying that config tweak burden?

Reply:
I think so. It would make buildkit more self-contained.

Reply:
I can dig in. Its turning into far more than the 5 lines I wanted :D

Reply:
Ended up with a full config-based implementation of hooks. Ill squash and force-push this commit with those details.